zbus/src/address/mod.rs

```diff
@@ -3,475 +3,123 @@
 //! comma-separated list of keys and values in the form key=value.
 //!
+//! # Miscellaneous and caveats on D-Bus addresses
+//!
+//! * Assumes values are UTF-8 encoded.
+//!
+//! * Duplicated keys are accepted, last pair wins.
+//!
+//! * Assumes that empty `key=val` is accepted, so `transport:,,guid=...` is valid.
+//!
+//! * Allows key only, so `transport:foo,bar` is ok.
+//!
+//! * Accept unknown keys and transports.
+//!
 //! See also:
 //!
 //! * [Server addresses] in the D-Bus specification.
 //!
 //! [Server addresses]: https://dbus.freedesktop.org/doc/dbus-specification.html#addresses
 
-pub mod transport;
+use core::fmt;
+use std::env;
 
-use crate::{Error, Guid, OwnedGuid, Result};
 #[cfg(all(unix, not(target_os = "macos")))]
 use nix::unistd::Uid;
-use std::{collections::HashMap, env, str::FromStr};
-
-use std::fmt::{Display, Formatter};
-
-use self::transport::Stream;
-pub use self::transport::Transport;
-
-/// A bus address.
-#[derive(Clone, Debug, PartialEq, Eq)]
-#[non_exhaustive]
-pub struct Address {
-    guid: Option<OwnedGuid>,
-    transport: Transport,
-}
-
-impl Address {
-    /// Create a new `Address` from a `Transport`.
-    pub fn new(transport: Transport) -> Self {
-        Self {
-            transport,
-            guid: None,
-        }
-    }
-
-    /// Set the GUID for this address.
-    pub fn set_guid<G>(mut self, guid: G) -> Result<Self>
-    where
-        G: TryInto<OwnedGuid>,
-        G::Error: Into<crate::Error>,
-    {
-        self.guid = Some(guid.try_into().map_err(Into::into)?);
 
-        Ok(self)
-    }
-
-    /// The transport details for this address.
-    pub fn transport(&self) -> &Transport {
-        &self.transport
-    }
-
-    #[cfg_attr(any(target_os = "macos", windows), async_recursion::async_recursion)]
-    pub(crate) async fn connect(self) -> Result<Stream> {
-        self.transport.connect().await
-    }
-
-    /// Get the address for the session socket respecting the `DBUS_SESSION_BUS_ADDRESS` environment
-    /// variable. If we don't recognize the value (or it's not set) we fall back to
-    /// `$XDG_RUNTIME_DIR/bus`.
-    pub fn session() -> Result<Self> {
-        match env::var("DBUS_SESSION_BUS_ADDRESS") {
-            Ok(val) => Self::from_str(&val),
-            _ => {
-                #[cfg(windows)]
-                return Self::from_str("autolaunch:");
-
-                #[cfg(all(unix, not(target_os = "macos")))]
-                {
-                    let runtime_dir = env::var("XDG_RUNTIME_DIR")
-                        .unwrap_or_else(|_| format!("/run/user/{}", Uid::effective()));
-                    let path = format!("unix:path={runtime_dir}/bus");
-
-                    Self::from_str(&path)
-                }
+pub mod transport;
 
-                #[cfg(target_os = "macos")]
-                return Self::from_str("launchd:env=DBUS_LAUNCHD_SESSION_BUS_SOCKET");
-            }
-        }
-    }
+#[allow(clippy::module_inception)]
+mod address;
```

Comment:
I was about to merge the PR and saw this. This file isn't large enough to justify making an exception here. The `Address` type doesn't need to move at all from here. It also makes the overall diff harder to read.